src/Transport/Config/AzureServiceBusTransportInfrastructure.cs

```diff
@@ -55,11 +55,21 @@ void InitializeIfNecessary()
 
             Guard.AgainstUnsetSerializerSetting(Settings);
 
+            var sendOnly = Settings.GetOrDefault<bool>("Endpoint.SendOnly");
+            var localAddressForSendOnlyEndpoint = ToTransportAddress(LogicalAddress.CreateLocalAddress(Settings.EndpointName(), new Dictionary<string, string>()));
+            try
+            {
+                localAddress = sendOnly ? localAddressForSendOnlyEndpoint : Settings.LocalAddress();
+            }
+            catch
```

Comment:
Is there a reason we are not catching `InvalidOperationException` here?

Reply:
As per discussion, this will only happen when core is not setting up the receive component. It means we're in a test that is not an acceptance test and we'll use the endpoint name as the local address (value assigned in case of a send only endpoint). Because of that, catching a specific exception is not required.

Reply:
Can't follow the reasoning why this catch is needed. Can you elaborate?

Reply:
@danielmarbach wasn't https://github.com/Particular/NServiceBus.AzureServiceBus/pull/666#issue-270455147 sufficient? (If not, I need to learn to explain things better 😞 ).

Reply:
Not really because it is confusing with the mentioning of send-only. Why isn't it sufficient to set in those tests the sendOnly flag? Or are we saying in these mentioned tests it is actually not a send only endpoint and thus we require this try catch?

Reply:
> Or are we saying in these mentioned tests it is actually not a send only endpoint and thus we require this try catch?

Precisely. It's a workaround.